deployments/apiv2/services/users/src/main.py

```diff
@@ -484,33 +499,52 @@ async def verify_user_account(
     """
     try:
         user_id = uuid.UUID(hex=token["userid"])
-        customer_id = uuid.UUID(hex=token["customer_id"])
+        is_customer = "customer:reset" in token["scope"]
+        is_user = "users" in token["scope"][0]
+        customer_id = None if is_customer else uuid.UUID(hex=token["customer_id"])
+
+        ph = PasswordHasher()
+        phash = ph.hash(details.password1.get_secret_value())
 
         async with request.state.pgpool.acquire() as con:
             async with con.transaction():
                 # ProtectedAny will return 401 already if link has expired
-                row = await con.fetchrow(
-                    "SELECT verified, pw_reset_verify_link FROM users WHERE id=$1 AND customer_id=$2",
-                    user_id,
-                    customer_id,
-                )
-                # if the link is being used to verify account and the account has already been verified, then return message to display to user
-                if details.verify and row["verified"]:
-                    return UnableToUpdateAccountResponse(message="Account has already been verified")
-                # link in db gets replaced with NULL when it's been successfully used
-                if row["pw_reset_verify_link"] is None:
-                    return UnableToUpdateAccountResponse(message="Link has already been used")
-
-                ph = PasswordHasher()
-                phash = ph.hash(details.password1.get_secret_value())
-
-                # will set verified to True even if it already is to remove extra, unnecessary conditional
-                await con.execute(
-                    "UPDATE users SET verified='t', pw_reset_verify_link=NULL, password=$1 WHERE id=$2 AND customer_id=$3",
-                    phash,
-                    user_id,
-                    customer_id,
-                )
+                if is_customer:
+                    row = await con.fetchrow(
+                        "SELECT pw_reset_link FROM customers WHERE id=$1",
+                        user_id,
+                    )
+                    # link in db gets replaced with NULL when it's been successfully used
+                    if row["pw_reset_link"] is None:
+                        return UnableToUpdateAccountResponse(message="Link has already been used")
+
+                    await con.execute(
+                        "UPDATE customers SET pw_reset_link=NULL, password=$1 WHERE id=$2",
+                        phash,
+                        user_id,
+                    )
+
+                elif is_user:
```

Comment:
Is there ever a situation where the account is not a user or customer? If it's always one of these two, you could just make this `else:` since you know it's a user if it's not a customer

Reply:
@tannermpeterson  in the resolved convo, jason mentioned wanting to prevent other types and to write it more defensively so only accept customer and users, don't use a catchall else statement